### DISCUSSION TITLE
seeking advice on a new adapter for doris  with incremental model

### DISCUSSION THREAD

**Discussion by catpineapple** [dbt-labs/docs.getdbt.com#1939] [Archive]
hi , i am trying to build a new adapter for apache doris,  But the implementation in incremental mode encountered some problems：
doris ：All import operations in Doris have atomicity guarantees, that is, the data in an import job either all succeed or all fail. It will not happen that only part of the data is imported successfully. 

for dbt-doris:
When multiple transactions are executed in parallel, only some of them succeed,  How should I design a solution: overall rollback or identify failed models for retry to improve the user's operating experience

Looking forward to your response.

**Comment by dataders** [comment]
hi @catpineapple I'm just seeing this now. you want to reach out in the #adapter-ecosystem channel in our community slack? also feel free to DM me there
